test-lit/semantics/mod-file-rewriter.f90

```diff
@@ -0,0 +1,9 @@
+! RUN: rm -r %t.dir && mkdir %t.dir && cd %t.dir
+! RUN: %f18 -fparse-only -fdebug-module-writer %s 2>&1 | FileCheck %s --check-prefix CHECK_FIRST_WRITE
+! RUN: %f18 -fparse-only -fdebug-module-writer %s 2>&1 | FileCheck %s --check-prefix CHECK_SECOND_WRITE
+
+module m
+end module m
+
+! CHECK_FIRST_WRITE: Processing module {{.*}}.mod: module written
+! CHECK_SECOND_WRITE: Processing module {{.*}}.mod: module unchanged, not writing
```

Comment:
A more thorough test would be to compile an initial version of `m`, then make a change that should result in the same .mod file and verify it is not rewritten. E.g. change:
```
real :: x(10)
```
to:
```
dimension :: x(10)
public :: x
real :: x
```
Then make a change that should result in a change (e.g. `public` to `private`) and verify it is rewritten.

Reply:
Thanks for the suggestion, I've changed the test to do exactly this